prompt2model/dataset_transformer/prompt_based.py

```diff
@@ -133,20 +109,97 @@ async def generate_responses(transform_prompts):
 
         try:
             loop = asyncio.get_event_loop()
-            responses = loop.run_until_complete(generate_responses(transform_prompts))
+            responses = loop.run_until_complete(
+                generate_responses_async(transform_prompts_batch)
+            )
         except API_ERRORS as e:
             handle_api_error(e)
+            # TODO: What to return here?
+        return responses
+
+    def process_responses(
+        self, responses: list, prompt_spec: PromptSpec
+    ) -> tuple[list[str], list[str]]:
+        """Process the responses received from the API.
+
+        Also write the current set of inputs and outputs to a dump text just in case.
 
+        Args:
+            responses: A list of response strings from the API.
+            prompt_spec: The PromptSpec object containing the instruction and examples.
+
+        Returns:
+            A tuple containing two lists: inputs and outputs.
+            - inputs: A list of transformed input strings.
+            - outputs: A list of transformed output strings.
+        """
+        inputs, outputs = [], []
+        show_sample_flag = True
         for response in responses:
             try:
                 extraction = find_and_parse_json(response, ["input", "output"], [])
                 if extraction is not None:
-                    inputs.append(str(extraction["input"]))
-                    outputs.append(str(extraction["output"]))
+                    if extraction["input"] is None or extraction["output"] is None:
+                        raise ValueError("Input or output is None")
+                    input = str(extraction["input"]).strip()
+                    output = str(extraction["output"]).strip()
+                    if input in prompt_spec.examples:
+                        raise ValueError("Repeated Task Examples from prompt")
+
+                    inputs.append(input)
+                    outputs.append(output)
+                    if show_sample_flag:
+                        logger.info(f"inputs\n{input}\n\nouputs\n{output}")
+                        show_sample_flag = False
+
             except Exception as e:
-                logger.error(f"Error extracting from response: {response}\nError: {e}")
-                continue
+                logger.error(f"Error extracting from response: {e}")
+                self.curr_failed_transforms += 1
+                if self.curr_failed_transforms > self.max_allowed_failed_transforms:
+                    break
+
+        with open("dump.txt", "a") as file:
+            file.write("Input: " + ", ".join(map(str, inputs)) + "\n")
+            file.write("Output: " + ", ".join(map(str, outputs)) + "\n")
```

Comment:
I think we can remove this

Reply:
Hmm, I'm a bit worried about transforms failing at large scale, this is a good intermediate but the logic for recovering data from dump isn't pushed so yeah will probably remove